code/datums/components/crafting/crafting.dm

```diff
@@ -392,17 +406,8 @@
 			var/mob/user = usr
 			var/datum/crafting_recipe/TR = locate(params["recipe"]) in GLOB.crafting_recipes
 			busy = TRUE
-			ui_interact(user)
-			var/atom/movable/result = construct_item(user, TR)
-			if(!istext(result)) //We made an item and didn't get a fail message
-				if(ismob(user) && isitem(result)) //In case the user is actually possessing a non mob like a machine
-					user.put_in_hands(result)
-				else
-					result.forceMove(user.drop_location())
-				to_chat(user, "<span class='notice'>[TR.name] constructed.</span>")
-			else
-				to_chat(user, "<span class='warning'>Construction failed[result]</span>")
-			busy = FALSE
+			. = TRUE
```

Comment:
Does it need to update if the construct_item_ui does the update?

Reply:
This update is so that the `busy` status kicks in immediately, locking the interface until crafting is completed. `construct_item_ui` can sleep and will only update at the end of its runtime, after any sleeps it might've done.

That said, it's a good point and I'll move it so `construct_item_ui` only does the update if crafting succeeded.

Reply:
Nevermind, since crafting can also fail at a delay, I think crafting would need a refactor to be able to only update when necessary. I think the current state is good enough - in the worst case, it updates the UI twice immediately when failing to craft, in the best case it updates UI once when starting to craft, and another once when ending, whether with success or failure.

Are there concerns that the worst case would allow you to spam two UI updates per click? If yes, maybe I could just add a little timeout after crafting failure before it finishes and unlocks the UI? I'm also adding a check to prevent starting another crafting while `busy`.